browser/models.py

```diff
@@ -3,83 +3,664 @@
-    numUpdates = db.Column(db.Integer, nullable=False, default=0)
-    firstUpdate = db.Column(db.TIMESTAMP)
-    lastUpdate = db.Column(db.TIMESTAMP)
 
-    def __init__(self, filename, state, subfolders):
-        self.filename = filename
-        self.state = state
-        self.subfolders = subfolders
+    raw_frames = db.relationship('RawFrame', backref='project')
+    rgb_frames = db.relationship('RGBFrame', backref='project')
+    label_frames = db.relationship('LabelFrame', backref='project')
+    # Includes underscore to avoid conflict with metadata (MetaData) attribute from SQLAlchemy
+    metadata_ = db.relationship('Metadata', backref='project', uselist=False)
+
+    def __init__(self, filename, input_bucket, output_bucket, path,
+                 rgb=False, raw_key=None, annotated_key=None):
+        init_start = timeit.default_timer()
+        if raw_key is None:
+            raw_key = 'raw'
+        if annotated_key is None:
+            annotated_key = get_ann_key(filename)
+
+        start = timeit.default_timer()
+        trial = self.load(filename, input_bucket, path)
+        current_app.logger.debug('Loaded file %s from S3 in %s s.',
+                                 filename, timeit.default_timer() - start)
+        raw = trial[raw_key]
+        annotated = trial[annotated_key]
+        # possible differences between single channel and rgb displays
+        if raw.ndim == 3:
+            raw = np.expand_dims(raw, axis=0)
+            annotated = np.expand_dims(annotated, axis=0)
+
+        # Create metadata
+        start = timeit.default_timer()
+        self.metadata_ = Metadata(self.id, filename, path, output_bucket,
+                                  raw, annotated, trial, rgb)
+        current_app.logger.debug('Created metadata for %s in %ss.',
+                                 filename, timeit.default_timer() - start)
+
+        # Create frames from raw, RGB, and labeled images
+        start = timeit.default_timer()
+        self.raw_frames = [RawFrame(self.id, i, frame)
+                           for i, frame in enumerate(raw)]
+        current_app.logger.debug('Created raw frames for %s in %ss.',
+                                 filename, timeit.default_timer() - start)
+
+        start = timeit.default_timer()
+        self.rgb_frames = [RGBFrame(self.id, i, frame)
+                           for i, frame in enumerate(raw)]
+        current_app.logger.debug('Created RGB frames for %s in %ss.',
+                                 filename, timeit.default_timer() - start)
+
+        start = timeit.default_timer()
+        self.label_frames = [LabelFrame(self.id, i, frame)
+                             for i, frame in enumerate(annotated)]
+        current_app.logger.debug('Created label frames for %s in %ss.',
+                                 filename, timeit.default_timer() - start)
+        # Log total time in constructor
+        current_app.logger.debug('Total time in project __init__ for %s: %s s.',
```

Comment:
Can we change the log text? Maybe something like `"Project for "%s" initialized in %s s."`?

Reply:
Changed to `'Initialized project for %s in %s s.'` to be most similar to the other log statements in Project `__init__`